src/main/java/gyro/azure/keyvault/VaultResource.java

```diff
@@ -0,0 +1,476 @@
+/*
+ * Copyright 2020, Perfect Sense, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package gyro.azure.keyvault;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import com.microsoft.azure.CloudException;
+import com.microsoft.azure.management.Azure;
+import com.microsoft.azure.management.keyvault.Vault;
+import com.microsoft.azure.management.resources.fluentcore.arm.Region;
+import gyro.azure.AzureResource;
+import gyro.azure.Copyable;
+import gyro.azure.resources.ResourceGroupResource;
+import gyro.core.GyroException;
+import gyro.core.GyroUI;
+import gyro.core.Type;
+import gyro.core.resource.Id;
+import gyro.core.resource.Output;
+import gyro.core.resource.Resource;
+import gyro.core.resource.Updatable;
+import gyro.core.scope.State;
+import gyro.core.validation.Required;
+
+/**
+ * The vault is managed using a combination of ``gyro`` configuration and commands. Currently ``certificate`` is the only entity that can be managed using ``gyro``.
```

Comment:
I don't think it's necessary to mark "gyro" as code. In fact, most uses of the word Gyro can probably be removed. I believe these docs can be line wrapped as well. It shouldn't need to be on a single line. That's only for getter docs.

Suggested wording:

```
 * Azure Key Vaults are managed using a combination resource configuration and commands. Currently 
 * ``aws::key-vault-certificate`` is the only entity that can be managed.
 *
 * Create an Azure Key Vault using the ``aws::key-vault`` resource. After the Key Vault is 
 * created use the ``gyro azure key-vault`` command to manage certificates within the key vault. 
 * See documentation below on how to create, add, or remove a certificate from a key vault.
```

Reply:
Refactored based on suggestion and correspondingly renamed the `vault` command to `key-vault` command